pkg/cvo/internal/generic_test.go

```diff
@@ -87,3 +87,71 @@ func TestCreateOnlyUpdate(t *testing.T) {
 		t.Error("should not have updated")
 	}
 }
+
+func TestCreateNoMatchError(t *testing.T) {
+	for _, tc := range []struct {
+		name     string
+		resource string
+	}{
+		{
+			name: "basic",
+			resource: `{
+  "kind": "UnknownType",
+  "apiVersion": "config.openshift.io/v1",
+  "metadata": {
+     "name": "cluster"
+  }
+}`,
+		},
+		{
+			name: "delete annotation",
+			resource: `{
+  "kind": "UnknownType",
+  "apiVersion": "config.openshift.io/v1",
+  "metadata": {
+     "name": "cluster",
+	 "annotations": {
+		 "release.openshift.io/delete": "true"
+	 }
+  }
+}`,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.Background()
+			obj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, []byte(tc.resource))
```

Comment:
you can probably avoid some of these runtime gymnastics by including `github.com/openshift/library-go/pkg/manifest` and using `json.Unmarshal` like we do [here][1].

[1]: https://github.com/openshift/cluster-version-operator/blob/887aa47093d740ce827e3046848d4f0c1aca374f/pkg/cvo/sync_test.go#L116-L118

Reply:
Updated as per recommendation.